### PR TITLE
Added redirect controller

### PR DESCRIPTION
**Pull request by dantleech**
This PR adds a dedicated `RedirectController` to handle AutoRoutes which have been assigned the `cmf_routing_auto.redirect` type.

| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | 
| License       | MIT
| Doc PR        | https://github.com/symfony-cmf/symfony-cmf-docs/pull/620

### CODE REVIEW

**Comment by dantleech** [comment]
@dbu ok with you? The tests look like a timeout issue, restarted.

File: .travis.yml [hunk 1]
```diff
@@ -19,6 +21,8 @@ matrix:
     - php: 5.5
       env: SYMFONY_VERSION=2.6.*
 
+allow
+
```
Comment (dbu) [inline .travis.yml]: what is this?

File: .travis.yml [hunk 2]
```diff
@@ -11,6 +11,8 @@ env:
   - SYMFONY_VERSION=2.5.*
 
 matrix:
+  allow_failures:
+    - php: hhvm
```
Comment (dbu) [inline .travis.yml]: why did hhvm start failing?

File: Resources/config/auto_route.xml [hunk 3]
```diff
@@ -86,6 +86,14 @@
             class="%cmf_routing_auto.metadata.factory.class%"
             factory-service="cmf_routing_auto.metadata.factory.builder"
             factory-method="getMetadataFactory"
-        />
+            />
+
+        <!-- Controller -->
+        <service 
+            id="cmf_routing_auto.redirect_controller"
+            class="Symfony\Cmf\Bundle\RoutingAutoBundle\Controller\RedirectController"
```
Comment (dbu) [inline Resources/config/auto_route.xml]: already following the new symfony best practices? but i agree - the mapping is done manually anyways in controllers_by_type.

File: Controller/RedirectController.php [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+namespace Symfony\Cmf\Bundle\RoutingAutoBundle\Controller;
+
+use Symfony\Cmf\Component\RoutingAuto\Model\AutoRouteInterface;
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\Routing\RouterInterface;
+
+/**
+ * Simple redirecting controller for AutoRouteInterface documents
```
Comment (dbu) [inline Controller/RedirectController.php]: can you explain here why this is a separate class? (we can't implement the RedirectRouteInterface)

File: Controller/RedirectController.php [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+    /**
+     * @param AutoRouteInterface $routeDocument
+     */
+    public function redirectAction(AutoRouteInterface $routeDocument)
+    {
+        $routeTarget = $routeDocument->getRedirectTarget();
+        $url = $this->router->generate($routeTarget);
+
+        return new RedirectResponse($url, 302);
```
Comment (dbu) [inline Controller/RedirectController.php]: should the redirect type be configureable between temporary and permanent?

Comment (dantleech) [inline Controller/RedirectController.php]: I think that would be a good idea

File: Resources/config/auto_route.xml (hunk 3 above)
Comment (dantleech) [inline Resources/config/auto_route.xml]: not intentionally :) I should probably change it to be consistent with everything else.

File: .travis.yml (hunk 2 above)
Comment (dantleech) [inline .travis.yml]: Not sure, but I have the feeling that hhvm is unstable and is always being a pain.

File: Resources/config/auto_route.xml (hunk 3 above)
Comment (dantleech) [inline Resources/config/auto_route.xml]: out of interest, where is that best practice defined ?

Comment (dbu) [inline Resources/config/auto_route.xml]: i think i read that in a blogpost on symfony.com but did not find it anymore. there is this https://github.com/symfony/symfony/issues/11881

Comment (WouterJ) [inline Resources/config/auto_route.xml]: It's here: http://symfony.com/doc/current/best_practices/business-logic.html#service-no-class-parameter

Anyway, that's going to be rewritten as the class parameters are going to be removed as there are better ways. See the issue @dbu referenced

File: .travis.yml (hunk 2 above)
Comment (WouterJ) [inline .travis.yml]: we should just add it as normal test and only do this when there is a problem with HHVM support.

File: Controller/RedirectController.php (hunk 5 above)
Comment (dantleech) [inline Controller/RedirectController.php]: We need to add this option to the schema, which is a bit more involved. I will create another ticket.

Comment (dantleech) [inline Controller/RedirectController.php]: https://github.com/symfony-cmf/RoutingAutoBundle/pull/146

**Comment by dantleech** [comment]
Hmm. Getting test failure from `security_authentication.utils` with symfony 2.3

Seems to be that we require the frameworkbundle v2.3 but have `symfony/symfony` at 2.6. Changing to require `symfony/symfony` SYMFONY_VERSION instead. /cc @WouterJ

File: Controller/RedirectController.php (hunk 4 above)
Comment (dantleech) [inline Controller/RedirectController.php]: Explainedg

**Comment by dantleech** [comment]
Shall we merge this now and I will add the backticks in another PR?

**Pull request merged**